### PR TITLE
Added examples folder in /client with two examples

### PR DESCRIPTION
**Pull request by sayden**
Continuing https://github.com/coreos/etcd/pull/3985

I have added two examples directory of client usage.
- "setGetKey" folder is the same example found in README.md with integration testing (and post cleaning) in "./test"
- "directory" folder is an example using directory listing and key retrieval with integration test (and post cleaning). Also integrated in global test "./test". Creates a "/myNodes" folder and added "key1" -> "value1" and "key2" -> "value2" to them.

Also, I have added in "./test" script the two folders with the examples so that they can be tested globally in integration. Everything written by the examples is later deleted and checked in the test.

I haven't touched the README.md as I wasn't sure what was the best way to update it with this information.

I hope this is the way you prefer.

### CODE REVIEW

File: client/examples/setGetKey/setGetKey.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		Transport: client.DefaultTransport,
+		// set timeout per request to fail fast when the target endpoint is unavailable
+		HeaderTimeoutPerRequest: time.Second,
+	}
+	c, err := client.New(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+	kapi := client.NewKeysAPI(c)
+	// set "/foo" key with "bar" value
```
Comment (jonboulle) [inline client/examples/setGetKey/setGetKey.go]: Please remove these comments, they're redundant with the messages being logged

File: client/examples/setGetKey/setGetKey.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	log.Print("Setting '/foo' key with 'bar' value")
+	resp, err := kapi.Set(context.Background(), "/foo", "bar", nil)
+	if err != nil {
+		log.Fatal(err)
+	} else {
```
Comment (jonboulle) [inline client/examples/setGetKey/setGetKey.go]: Remove the else - prefer [short returns](https://golang.org/doc/effective_go.html#if)

File: client/examples/setGetKey/setGetKey.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		// print common key info
+		log.Printf("Set is done. Metadata is %q\n", resp)
+	}
+	// get "/foo" key's value
+	log.Print("Getting '/foo' key value")
+	resp, err = kapi.Get(context.Background(), "/foo", nil)
+	if err != nil {
+		log.Fatal(err)
+	} else {
```
Comment (jonboulle) [inline client/examples/setGetKey/setGetKey.go]: Remove else

File: client/examples/directory/directory_test.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+func TestMain(m *testing.M) {
+	main()
+	code := m.Run()
+
+	//Clean
+	log.Println("Cleaning test...")
+
+	//Clean nodes within "myNodes
+	cfg := client.Config{
```
Comment (jonboulle) [inline client/examples/directory/directory_test.go]: This isn't good, we can't expect there to be an etcd running and do arbitrary things to it. Can you instead use [integration.NewCluster](https://github.com/coreos/etcd/blob/master/integration/cluster_test.go#L393), as [Xiang suggested](https://github.com/coreos/etcd/pull/3985#issuecomment-164506812)?

**Comment by heyitsanthony** [comment]
Superseded by #8034. Closing

**Pull request closed without merging**